examples/keyboard/Keyboard.cpp

```diff
@@ -0,0 +1,855 @@
+        CASE(Right);
+        CASE(Left);
+        CASE(Down);
+        CASE(Up);
+        CASE(NumLock);
+        CASE(NumpadDivide);
+        CASE(NumpadMultiply);
+        CASE(NumpadMinus);
+        CASE(NumpadPlus);
+        CASE(NumpadEqual);
+        CASE(NumpadEnter);
+        CASE(NumpadDecimal);
+        CASE(Numpad1);
+        CASE(Numpad2);
+        CASE(Numpad3);
+        CASE(Numpad4);
+        CASE(Numpad5);
+        CASE(Numpad6);
+        CASE(Numpad7);
+        CASE(Numpad8);
+        CASE(Numpad9);
+        CASE(Numpad0);
+        CASE(NonUsBackslash);
+        CASE(Application);
+        CASE(Execute);
+        CASE(ModeChange);
+        CASE(Help);
+        CASE(Menu);
+        CASE(Select);
+        CASE(Redo);
+        CASE(Undo);
+        CASE(Cut);
+        CASE(Copy);
+        CASE(Paste);
+        CASE(VolumeMute);
+        CASE(VolumeUp);
+        CASE(VolumeDown);
+        CASE(MediaPlayPause);
+        CASE(MediaStop);
+        CASE(MediaNextTrack);
+        CASE(MediaPreviousTrack);
+        CASE(LControl);
+        CASE(LShift);
+        CASE(LAlt);
+        CASE(LSystem);
+        CASE(RControl);
+        CASE(RShift);
+        CASE(RAlt);
+        CASE(RSystem);
+        CASE(Back);
+        CASE(Forward);
+        CASE(Refresh);
+        CASE(Stop);
+        CASE(Search);
+        CASE(Favorites);
+        CASE(HomePage);
+        CASE(LaunchApplication1);
+        CASE(LaunchApplication2);
+        CASE(LaunchMail);
+        CASE(LaunchMediaSelect);
+#undef CASE
+    }
+
+    throw std::runtime_error("invalid keyboard scancode");
+}
+
+////////////////////////////////////////////////////////////
+// Entity showing keyboard events and real-time state on a keyboard
+////////////////////////////////////////////////////////////
+class KeyboardView : public sf::Drawable, public sf::Transformable
+{
+public:
+    explicit KeyboardView(const sf::Font& font) :
+    m_triangles(sf::PrimitiveType::Triangles, sf::Keyboard::ScancodeCount * 6),
+    m_labels(sf::Keyboard::ScancodeCount, sf::Text(font, "", 14))
+    {
+        // Check all the scancodes are in the matrix exactly once
+        {
+            std::unordered_set<sf::Keyboard::Scancode> scancodesInMatrix;
+            for (const auto& [cells, marginBottom] : m_matrix)
+            {
+                for (const auto& [scancode, size, marginRight] : cells)
+                {
+                    assert(scancodesInMatrix.count(scancode) == 0);
+                    scancodesInMatrix.insert(scancode);
+                }
+            }
+            assert(scancodesInMatrix.size() == sf::Keyboard::ScancodeCount);
+        }
+
+        // Initialize keys color and label
+        forEachKey(
+            [this](sf::Keyboard::Scancode scancode, const sf::FloatRect& rect)
+            {
+                const auto scancodeIndex = static_cast<std::size_t>(scancode);
+
+                for (std::size_t vertexIndex = 0u; vertexIndex < 6u; vertexIndex++)
```

Comment:
It really doesn't matter, but ok.

Reply:
https://github.com/SFML/SFML/pull/2019

Here is the PR that established this precedent. I agree with Vittorio's reasoning that because pre-increment is simpler, we ought to use it when possible. Only use post-increment when we need the extra functionality it provides. I agree it's a minuscule change but I still think it's worthwhile to maintain this precedent.